xarray/namedarray/pycompat.py

```diff
@@ -145,3 +145,28 @@ def to_duck_array(data: Any, **kwargs: dict[str, Any]) -> duckarray[_ShapeType,
         return data
     else:
         return np.asarray(data)  # type: ignore[return-value]
+
+
+async def async_to_duck_array(
+    data: Any, **kwargs: dict[str, Any]
+) -> duckarray[_ShapeType, _DType]:
+    from xarray.core.indexing import (
+        ExplicitlyIndexed,
+        ImplicitToExplicitIndexingAdapter,
+    )
+    from xarray.namedarray.parallelcompat import get_chunked_array_type
+
+    print(type(data))
+
+    if is_chunked_array(data):
+        chunkmanager = get_chunked_array_type(data)
+        loaded_data, *_ = chunkmanager.compute(data, **kwargs)  # type: ignore[var-annotated]
+        return loaded_data
+
+    if isinstance(data, ExplicitlyIndexed | ImplicitToExplicitIndexingAdapter):
```

Comment:
I think we should keep this if, and then in the else call `return to_duck_array(data)`